### PR TITLE
Fixed the issue with Inclusion/Exclusion for the Encryption to support the Normal and Multipart request 

### PR DESCRIPTION
**Pull request by Parimal-shaw**
The issue with the inclusion and exclusion of parameters during encryption for normal and multipart HTTP requests in both the "**Parameter value**" and "**Parameter Key and value**" modes has been fixed. In short, PyCript now supports the inclusion and exclusion of parameters during encryption for most types of HTTP requests.

### CODE REVIEW

**Comment by Parimal-shaw** [comment]
New commit was made to fix the conflict.

File: pycript/Reqcheck.py [hunk 1]
```diff
@@ -189,55 +189,63 @@ def decrypt_and_update_parameter_keys_and_values(extender, currentreq, decryptio
...
                 currentreq = extender.helpers.updateParameter(currentreq, extender.helpers.buildParameter(param.getName(), encrypted_param_value, param.getType()))
+                
             elif param.getType() == IParameter.PARAM_JSON:
                 json_object = loads(body)
                 json_object = update_json_value(json_object, selectedlang, encryptionpath, Parameterencrypt,selected_request_inc_ex_ctype,listofparam)
                 output = extender.helpers.stringToBytes(dumps(json_object,separators=(',', ':')))
                 currentreq = extender.helpers.buildHttpMessage(header, output)
                 break
         
+            else:
```
Comment (Anof-cyber) [inline pycript/Reqcheck.py]: ` if param.getType() == IParameter.PARAM_URL:` added inside the `elif selected_method == "BODY" and param.getType() != IParameter.PARAM_URL:`  


 `elif selected_method == "BODY" and param.getType() != IParameter.PARAM_URL:`   will always exclude the get parameter, means ` if param.getType() == IParameter.PARAM_URL:` is never going to trigged and get parameter will be excluded 
 
 ` if param.getType() == IParameter.PARAM_URL:` should not be part of the another nested condition

File: pycript/Reqcheck.py [hunk 2]
```diff
@@ -189,55 +189,63 @@ def decrypt_and_update_parameter_keys_and_values(extender, currentreq, decryptio
...
         if selected_method == "GET" and param.getType() == IParameter.PARAM_URL:
-            encrypted_param_value = Parameterencrypt(selectedlang, encryptionpath, param.getValue())
+            parameter_value = param
+            encrypted_param_value = update_raw_value(parameter_value, selectedlang, encryptionpath, Parameterencrypt,selected_request_inc_ex_ctype, listofparam)
             currentreq = extender.helpers.updateParameter(currentreq, extender.helpers.buildParameter(param.getName(), encrypted_param_value, param.getType()))
-
         elif selected_method == "BODY" and param.getType() != IParameter.PARAM_URL:
             if param.getType() == IParameter.PARAM_BODY:
-                encrypted_param_value = Parameterencrypt(selectedlang, encryptionpath, param.getValue())
+                parameter_value = param
```
Comment (Anof-cyber) [inline pycript/Reqcheck.py]: `update_raw_value` and `update_raw_key_value` both take `Parameterencrypt` function as arg?

`Parameterencrypt` is indeed a separate function passing a func to another func reduce the executive speed at certain level due to nested conditions, 

both update function can directly call the `Parameterencrypt` function, arg is not required

File: pycript/utils.py [hunk 3]
```diff
@@ -88,6 +88,69 @@ def update_json_key_value(json_obj, selectedlang, decryptionpath,enc_dec ,select
...
+    param_name = parameter_value.getName()
+    param_value = parameter_value.getValue()
+    
+    if selected_request_response_inc_ex_ctype is None:
+        # Process all parameters in the parameter_valueect
+        new_value = enc_dec(selectedlang, encryptionpath, param_value)
+
+    else:
+        # Process parameters based on selectedparamtertype
+        if selected_request_response_inc_ex_ctype == "Include Parameters":
```
Comment (Anof-cyber) [inline pycript/utils.py:102]: we can have separate if condition, or if elif else. Adding else with the same if else condition make is difficult to read and not frndly

File: pycript/execution.py [hunk 4]
```diff
@@ -16,6 +16,7 @@ def execute_command(selectedlang, path, data, headervalue=None):
         with tempfile.NamedTemporaryFile(delete=False, mode='w') as temp_file:
             json.dump(content, temp_file)
             temp_file_path = temp_file.name
+            print(temp_file_path)
```
Comment (Anof-cyber) [inline pycript/execution.py]: unwanted, temp files are always deleted during execution process, printing file name make no sense and not useful as after the print and execution file is already deleted

**Review by Anof-cyber — APPROVED** [review]

**Comment by Anof-cyber** [comment]
Removed the changes for `Reqcheck.py` due to multiple logical issues and unwanted code.

`execution.py`  - [Need need for](https://github.com/Anof-cyber/PyCript/pull/15#discussion_r1885069594)

**Pull request merged**